Doc provisoin script improvemnts (#283)
nrfconnect/sdk-sidewalk: doc/setting_up_sidewalk_environment/setting_up_sidewalk_product.rst

Setting up your Sidewalk product
################################

To complete the set up of your Sidewalk product, you have to generate provisioning, and add MQTT to destination.

Preconditions
*************

Before creating a Sidewalk provisioning file, you have to:

 * Create an AWS account
 * Set up an AWS user with permissions to create resources
 * Set up users aws *credentials* file on your local machine

> For detailed instructions see `Amazon Sidewalk IoT Prerequisites`_ (you just need to follow the `Prerequisites` section).

Reviewer: this one it still valid

> For detailed instructions see `Amazon Sidewalk IoT Prerequisites`_ (you just need to follow the `Prerequisites` section).

Author: I think we should leave the link to the prerequisites article

Setting up your Sidewalk product
################################

To complete the set up of your Sidewalk product, you have to generate provisioning and add MQTT to destination.

Preconditions
*************

Before creating a Sidewalk provisioning file, you have to:

 * Create an AWS account
 * Set up an AWS user with permissions to create resources
 * Set up user's AWS credentials file on your local machine

>  * Complete the `Amazon Sidewalk IoT Prerequisites`_  instructions

Reviewer: ```suggestion
 * Complete the `Amazon Sidewalk IoT Prerequisites`_ instructions
```

#. Run device provisioning scripts:

>    a. Setup Python virtual environment for the provisioning tools
> 
>       .. code-block:: console
> 
>          $ python3 -m pip install --user virtualenv
>          $ python3 -m venv sample-app-env
>          $ source sample-app-env/bin/activate
>          (sample-app-env)$ python3 -m pip install --upgrade pip
>          (sample-app-env)$ python3 -m pip install -r requirements.txt

Reviewer 2: ```suggestion
   a. Setup Python virtual environment for the provisioning tools
      
      * Linux:

      .. code-block:: console

         python3 -m pip install --user virtualenv
         python3 -m virtualenv sample-app-env
         source sample-app-env/bin/activate
         python3 -m pip install --upgrade pip
         python3 -m pip install -r requirements.txt
         python3 -m pip install pyjwt -t ./ApplicationServerDeployment/lambda/authLibs
         
      * Windows:
      
      .. code-block:: console

         python3 -m pip install --user virtualenv
         python3 -m virtualenv sample-app-env
         sample-app-env\Scripts\activate.bat
         python3 -m pip install --upgrade pip
         python3 -m pip install -r requirements.txt
         python3 -m pip install pyjwt -t ./ApplicationServerDeployment/lambda/authLibs
```

   #. Run device provisioning scripts:

      .. code-block:: console
   

>          (sample-app-env)$ python3 EdgeDeviceProvisioning/provision_sidewalk_end_device.py

Reviewer 2: ```suggestion
         python3 EdgeDeviceProvisioning/provision_sidewalk_end_device.py
```

Author: I would like to mark, this commands must be run in virtual env

      You should see the following output:

      .. code-block:: console

         INFO:root:Status: 200
         INFO:root:Saving wireless device to file
         INFO:root:Generating MFG by calling provision.py
         INFO:root:  Generating MFG.hex for Nordic
         INFO:root:Done!

   #. Exit Python virtual environment

      .. code-block:: console

>          (sample-app-env)$ deactivate

Reviewer 2: ```suggestion
         deactivate
```

#. Flash the :file:`Nordic_MFG.hex` file:

>    Your provisioning file is located in `EdgeDeviceProvisioning` folder.

Reviewer 3: ```suggestion
   Your provisioning file is located in `EdgeDeviceProvisioning` directory.
```

>    Devices are grouped in device profiles folders as shown in structure below:

Reviewer 3: ```suggestion
   Devices are grouped in device profiles subdirectory as shown in structure below:
```

   .. code-block:: console

      EdgeDeviceProvisioning \
      - DeviceProfile_<profile-id> \
         - DeviceProfile.json
         - WirelessDevice_<device-id>\
             --  Nordic_MFG.bin
             --  Nordic_MFG.hex
             --  WirelessDevice.json

>    a. Go to device folder

Reviewer 3: ```suggestion
   a. Go to device subdirectory
```

      .. code-block:: console

>          $ cd EdgeDeviceProvisioning/DeviceProfile_<profile-id>/WirelessDevice_<device-id>

Reviewer 2: ```suggestion
         cd EdgeDeviceProvisioning/DeviceProfile_<profile-id>/WirelessDevice_<device-id>
```

      For example:

      .. code-block:: console

>          $ cd EdgeDeviceProvisioning/DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9/WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911

Reviewer 3: I have a proposal to add additional info that the pattern after `WirelessDevice_` subcatalog name is a device ID which can be used with MQTT to send messages to device. It can be helpful for user.

Author: Added a reference in testing document. I point to the `WirelessDevice.json` file instead of folder name

>          $ cd EdgeDeviceProvisioning/DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9/WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911

Reviewer 2: ```suggestion
         cd EdgeDeviceProvisioning/DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9/WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911
```

   #. Flash .hex file with provisioning data: 

      .. code-block:: console

>          $ nrfjprog --sectorerase --program Nordic_MFG.hex --reset

Reviewer 2: ```
   #. Flash .hex file with provisioning data: 

      .. code-block:: console
         nrfjprog --sectorerase --program Nordic_MFG.hex --reset

       .. note::
          
          If you re-flashed mfg.hex file on already working device, please make sure to perform factory reset (long BUTTON_1 press) to de-register previously flashed device to allow new product (new mfg.hex) register in Sidewalk network
```

Author: Added note

         INFO:root:Done!

   #. Exit the Python virtual environment:

      .. code-block:: console

         deactivate

#. Flash the :file:`Nordic_MFG.hex` file.

   Your provisioning file is located in the :file:`EdgeDeviceProvisioning` directory.
   Devices are grouped in the device profile's subdirectory as shown in the structure below:

   .. code-block:: console

      EdgeDeviceProvisioning \
      - DeviceProfile_<profile-id> \
         - DeviceProfile.json
         - WirelessDevice_<device-id>\
             --  Nordic_MFG.bin
             --  Nordic_MFG.hex
             --  WirelessDevice.json

   a. Go to the device subdirectory:

      .. code-block:: console

         cd EdgeDeviceProvisioning/DeviceProfile_<profile-id>/WirelessDevice_<device-id>

      For example:

      .. code-block:: console

         cd EdgeDeviceProvisioning/DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9/WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911

   #. Flash the :file:`Nordic_MFG.hex` file with the provisioning data: 

      .. code-block:: console

         $ nrfjprog --sectorerase --program Nordic_MFG.hex --reset

      .. note::

>          If you reflashed the :file:`Nordic_MFG.hex` file on an already working device, make sure to perform a factory reset (**Button 1**  long press) to deregister the previously flashed device.

Reviewer: ```suggestion
         If you reflashed the :file:`Nordic_MFG.hex` file on an already working device, make sure to perform a factory reset (**Button 1** long press) to deregister the previously flashed device.
```
